raiden/smart_contracts/TokenNetwork.sol

```diff
@@ -0,0 +1,697 @@
+
+        // Store channel information
+        channels[last_channel_index] = Channel({settle_timeout: settle_timeout});
+
+        // Mark the channel participants
+        // We use this in setDeposit to ensure the beneficiary is a channel participant
+        channels[last_channel_index].participants[participant1].initialized = true;
+        channels[last_channel_index].participants[participant2].initialized = true;
+
+        ChannelOpened(last_channel_index, participant1, participant2, settle_timeout);
+
+        return last_channel_index;
+    }
+
+    /// @notice Sets the channel participant total deposit value.
+    /// Can be called by anyone.
+    /// @param channel_identifier The channel identifier - mapping key used for `channels`
+    /// @param participant Channel participant who's deposit is being set.
+    /// @param total_deposit Idempotent function which sets the total amount of tokens that the participant will have as a deposit.
+    function setDeposit(
+        uint256 channel_identifier,
+        address participant,
+        uint256 total_deposit)
+        public
+    {
+        uint256 added_deposit;
+        Channel storage channel = channels[channel_identifier];
+
+        // Channel must be open and participant must be part of the channel.
+        require(channel.participants[participant].initialized);
+
+        // Channel cannot be closed
+        require(closing_requests[channel_identifier].settle_block_number == 0);
+
+        require(channel.participants[participant].deposit < total_deposit);
+
+        added_deposit = total_deposit - channel.participants[participant].deposit;
+
+        // Sender should have enough balance
+        require(token.balanceOf(msg.sender) >= added_deposit);
+
+        // Change the state
+        channel.participants[participant].deposit += added_deposit;
+
+        // Do the transfer
+        require(token.transferFrom(msg.sender, address(this), added_deposit));
+
+        ChannelNewDeposit(channel_identifier, participant, channel.participants[participant].deposit);
+    }
+
+    /// @notice Close a channel between two parties that was used bidirectionally.
+    /// Only a participant may close the channel, providing a balance proof signed by its partner. Callable only once.
+    /// @param channel_identifier The channel identifier - mapping key used for `channels`
+    /// @param nonce Strictly monotonic value used to order transfers.
+    /// @param transferred_amount Total amount of tokens transferred by the channel partner
+    /// to the channel participant who calls the function.
+    /// @param locksroot Root of the partner's merkle tree of all pending lock lockhashes.
+    /// @param additional_hash Computed from the message. Used for message authentication.
+    /// @param signature Partner's signature of the balance proof data.
+    function closeChannel(
+        uint256 channel_identifier,
```

Comment:
This is missing the token address. It must be added and `require`d, otherwise a balance proof from a different network with the same participants can be used here.

It's also missing the chain_id

Reply:
Yes, the `chain_id` is missing. I will add it.
I used the `TokenNetwork` contract address instead of the token address. This is what we agreed on.